File: 02_algorithm/programmers/Level1/하샤드 수.py
def solution(x):
    Sum_value = 0
    for digit in str(x):
        Sum_value += int(digit)
    if not x % Sum_value:
        answer = True
    else:
        answer = False
    return answer

print(10)
print(18)
print(11)
print(12)

File: 06_js/04_algorithm/programmers/Level1/하샤드 수.js
function solution(x) {
  let str_num = x.toString()
  let sum_value = 0
  for (let i = 0; i < str_num.length; i++) {
      sum_value += parseInt(str_num[i])
  }
  if (x % sum_value === 0) {
      return true
  } else {
      return false
  }
}

console.log(solution(10))
console.log(solution(11))
console.log(solution(12))
console.log(solution(13))
console.log(solution(144))